### PR TITLE
Remove all coupons before each test run, and clear cart

### LINKED ISSUE #29437: Empty cart before E2E cart related tests
When one of the early cart/checkout coupon tests leaves a coupon in the cart, it breaks the coupon tests that follow. The cart should be emptied at the beginning of each cart/checkout test. 

A secondary consideration would be deleting all existing coupons before running coupon related tests. This could be done using the API as we currently do with products.

screenshot_of_failed_test

### PR DESCRIPTION
**Pull request by zhongruige**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/29437.

### How to test the changes in this Pull Request:

1. Spin up the e2e Docker environment with `npx wc-e2e docker:up`
2. Login at `http://localhost:8084/wp-login.php` and create some coupons
3. Open the coupons page and note the coupons you created, for example:

Screen Shot 2021-03-30 at 3 09 37 PM

4. Run either a single test or the entire e2e test suite
5. Once the test starts, refresh the page and verify the coupons are no more:

Screen Shot 2021-03-30 at 3 10 00 PM

6. Make sure the tests run and pass locally as well as in CI

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

n/a

### CODE REVIEW

File: tests/e2e/utils/src/flows/shopper.js [hunk 1]
```diff
@@ -127,6 +129,33 @@ const shopper = {
...
+					await uiUnblocked();
+				}
+				products = await page.$( '.remove' );
+			}
+		}
+
+		// Remove coupons if they exist
+		if ( await page.$( '.woocommerce-remove-coupon' ) !== null ) {
+			await page.click( '.woocommerce-remove-coupon' );
+			await uiUnblocked()
```
Comment (rrennick) [inline tests/e2e/utils/src/flows/shopper.js:152]: Nitpick- this is missing the trailing semi-colon.

**Review by rrennick — CHANGES_REQUESTED** [review]
@zhongruige Nice work on this. Tests passed locally:

```
Test Suites: 5 passed, 5 total
Tests:       2 skipped, 141 passed, 143 total
Snapshots:   0 total
Time:        576.015s
Ran all test suites.
```

I'll need to re-review after addressing the nitpick. Otherwise this is ready for merge.

File: tests/e2e/utils/src/flows/shopper.js (hunk 1 above)
Comment (zhongruige) [reply tests/e2e/utils/src/flows/shopper.js:152]: Ack, good catch. Fixed!

**Review by rrennick — APPROVED** [review]
@zhongruige Nice work on this.

**Pull request merged**